strawberryfields/configuration.py

```diff
@@ -222,6 +295,81 @@ def parse_environment_variable(key, value):
     return value
 
 
+def active_configs(filename="config.toml"):
+    """Prints the filepaths for existing configuration files to the standard
+    output and marks the one that is active.
+
+    This function relies on the precedence ordering of directories to check
+    when marking the active configuration.
+
+    Args:
+        filename (str): the name of the configuration files to look for
+    """
+    active_configs_list = get_active_configs(filename)
+    print_active_configs(active_configs_list, filename)
+    print_directories_checked()
+
+
+def get_active_configs(filename="config.toml"):
+    """
+    Args:
+        filename (str): the name of the configuration files to look for
+
+    Returns:
+        list[str]: the filepaths for the active configurations
+    """
+    active_configs_list = []
+
+    directories = directories_to_check()
+
+    for directory in directories:
+        filepath = os.path.join(directory, filename)
+        if os.path.exists(filepath):
+            active_configs_list.append(filepath)
+
+    return active_configs_list
+
+
+def print_active_configs(active_configs_list, filename):
+    """Prints the active configurations found based on the filename specified.
+
+    This function relies on the precedence ordering of directories to mark the
+    active configuration.
+
+    Args:
+        active_configs_list (list[str]): the filepaths for the active configurations
+        filename (str): the name of the configuration file
+    """
+    if active_configs_list:
+        active = True
+
+        print(
+            "\nThe following Strawberry Fields configuration files were found "
+            'with the name "{}":\n'.format(filename)
+        )
+
+        for config in active_configs_list:
+            if active:
+                config += " (active)"
+                active = False
+
+            print("* " + config)
+    else:
+        print(
+            "\nNo Strawberry Fields configuration files were found with the "
+            'name "{}".\n'.format(filename)
+        )
+
+
+def print_directories_checked():
+    """Prints the directores that are being checked for a configuration file."""
+    directories = directories_to_check()
+
+    print("\nThe following directories were checked:\n")
+    for directory in directories:
+        print("* " + directory)
```

Comment:
Nitpick: I see the logic behind splitting out the two print into separate functions (functions should do one thing, easier testing). In this case, however, I feel like the re-use capability for these two subfunctions is pretty low (since they do not return anything, just print), and find it a bit easier to follow the logic if they were part of `active_configs`.

Reply:
It seems something like:
```python
    active_configs = get_active_configs(filename)

    if active_configs:
        active = True

        print("\nThe following Strawberry Fields configuration files were found "
              "with the name \"{}\":\n".format(filename))

        for config in active_configs:
            if active:
                config += " (active)"
                active = False

            print("* " + config)
    else:
        print("\nNo Strawberry Fields configuration files were found with the "
              "name \"{}\".\n".format(filename))

    directories = directories_to_check()

    print("\nThe following directories were checked:\n")
    for directory in directories:
        print("* " + directory)
```

I'd be in for having the transition, as it is still not entirely difficult to parse. @thisac what are your thoughts on this?

Reply:
Sure, I think that it makes sense to have them in the same function actually, even though it's sort of the opposite of what was told during the clean-code seminar. I would consider adding some comments though detailing what's being done in the different parts of the function.